osu.Game/Overlays/Profile/UserGraph.cs

```diff
@@ -0,0 +1,284 @@
+            private readonly Box ballBg;
+            private readonly Box line;
+
+            public Action<int> OnBallMove;
+
+            public UserLineGraph()
+            {
+                Add(bar = new Container
+                {
+                    Origin = Anchor.TopCentre,
+                    RelativeSizeAxes = Axes.Y,
+                    AutoSizeAxes = Axes.X,
+                    Alpha = 0,
+                    RelativePositionAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        line = new Box
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            RelativeSizeAxes = Axes.Y,
+                            Width = 2,
+                        },
+                        movingBall = new CircularContainer
+                        {
+                            Anchor = Anchor.TopCentre,
+                            Origin = Anchor.Centre,
+                            Size = new Vector2(20),
+                            Masking = true,
+                            BorderThickness = 4,
+                            RelativePositionAxes = Axes.Y,
+                            Child = ballBg = new Box { RelativeSizeAxes = Axes.Both }
+                        }
+                    }
+                });
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OverlayColourProvider colourProvider, OsuColour colours)
+            {
+                ballBg.Colour = colourProvider.Background5;
+                movingBall.BorderColour = line.Colour = colours.Yellow;
+            }
+
+            public void UpdateBallPosition(float mouseXPosition)
+            {
+                const int duration = 200;
+                int index = calculateIndex(mouseXPosition);
+                Vector2 position = calculateBallPosition(index);
+                movingBall.MoveToY(position.Y, duration, Easing.OutQuint);
+                bar.MoveToX(position.X, duration, Easing.OutQuint);
+                OnBallMove.Invoke(index);
+            }
+
+            public void ShowBar() => bar.FadeIn(FADE_DURATION);
+
+            public void HideBar() => bar.FadeOut(FADE_DURATION);
+
+            private int calculateIndex(float mouseXPosition) => (int)MathF.Round(mouseXPosition / DrawWidth * (DefaultValueCount - 1));
+
+            private Vector2 calculateBallPosition(int index)
+            {
+                float y = GetYPosition(Values.ElementAt(index));
+                return new Vector2(index / (float)(DefaultValueCount - 1), y);
+            }
+        }
+
+        protected abstract class UserGraphTooltip : VisibilityContainer, ITooltip
+        {
+            protected readonly OsuSpriteText Counter, BottomText;
+            private readonly Box background;
+
+            /// <summary>
+            /// Text which will be shown near the <see cref="Counter"/>.
```

Comment:
This xmldoc is very nondescript and doesn't really say what this *does*. Better:
```suggestion
            /// Text describing the value being plotted on the graph, which will be displayed as a prefix to the value of <see cref="Counter"/>.
```